prototype2/adc_readout/AdcParse.h

```diff
@@ -49,6 +49,19 @@ class ParserException : public std::runtime_error {
   std::string Error;
 };
 
+struct ChannelID {
+  std::uint16_t ChannelNr;
+  std::uint16_t SourceID;
+  bool operator==(ChannelID const &Other) const {
+    return Other.ChannelNr == ChannelNr and Other.SourceID == SourceID;
+  };
+  bool operator<(ChannelID const &Other) const {
+    const auto MaxNrOfChannels = 4;
```

Comment:
Does ChannelID  always contain valid values? Should there be out of range checks in both operator == and < ? 

Why define MaxNrOfChannels here - I would assume that it was a key constraint to be set in constructor?

Reply:
I do not want the maximum number of channels to be a member variable, hence why it is defined in the `operator<()` function. The existence of the two comparison functions is in order to enable sorting. I will add default init values to ChannelID.